<!-- -->
<launch>

    <!-- initialize the gridmap - provide dimension and resolution here -->
    <!-- This is the structure for shared memory being used by
    different grid based nodes. should be called only once -->
    <node pkg="gridmapping2" type="gridmapping2_shared_gridmap_node" name="shared_gridmap" output="screen">
      <param name="gridDimensionX" value="160"/>
      <param name="gridDimensionY" value="160"/>
      <param name="gridDimensionZ" value="50"/>
      <param name="gridResolution" value=".1"/>
      <param name="gridIdentifier" value="shared_grid_map"/>
</node>


    <!--Add point cloud to rgbd grid -->
    <!--Needs a transform to exist between worldframe and
    pointcloudframe, pointcloudframe doesnt need be the same as actual
    frame_id of point_cloud-->
    <node pkg="rgbd_grid_sensor" type="rgbd_grid_sensor_node" name="rgbd_grid_sensor" output="screen" >
      <remap from="/rgbd_grid_sensor/point_cloud" to="/point_cloud"/> 
      <param name="pointCloudFrame" value="/camera_rgb_optical_frame" />
      <param name="worldFrame" value="/world" />
    </node>
    
    <!-- params required by scrolling strategies (read in
    scrollgrid/scroll_strategies-->
    <param name="scrolling_strategies/scroll_dist_thresh_x" value="10" />
    <param name="scrolling_strategies/scroll_dist_thresh_y" value="10" />
    <param name="scrolling_strategies/scroll_dist_thresh_z" value="5" />
    <param name="scrolling_strategies/target_sensor_to_center_x" value="0"/>
    <param name="scrolling_strategies/target_sensor_to_center_y" value="0"/>
    <param name="scrolling_strategies/target_sensor_to_center_z" value="0"/>
    <!--Scroll strategy class is called in rgbd_grid_sensor but
    currently takes in parameters independently -->

    <!-- to visualize use rgbd_grid_visualization-->

</launch>
